src/App.tsx

```diff
@@ -15,35 +16,116 @@
   'Garlic',
 ];
 
+
+
 export const App: React.FC = () => {
+  const [sortedGoods, setSortedGoods] = useState([...goodsFromServer]);
+  const [reversePressed, setReversePressed] = useState(false);
+  const [alphabeticallyPressed, setAlphabetically] = useState(false);
+  const [byLengthPressed, setByLength] = useState(false);
+
+  const sortGoods = (sortType: string) => {
+    let sorted = [...sortedGoods];
+
+    switch (sortType) {
+      case 'alphabetically':
+        sorted.sort((a, b) => {
+          setAlphabetically(!alphabeticallyPressed);
+          if (byLengthPressed) {
+            setByLength(!byLengthPressed);
+          }
+          if (reversePressed) {
+            return a.localeCompare(b) * -1;
+          } else {
+            return a.localeCompare(b)
+          }
+        })
+        break;
+
+      case 'byLength':
+        sorted.sort((a, b) => {
+          setByLength(!byLengthPressed);
+          if (alphabeticallyPressed) {
+            setAlphabetically(!alphabeticallyPressed);
+          }
+          if (reversePressed) {
+            return (a.length - b.length) * -1;
+          } else {
+            return a.length - b.length
+          }
+        })
```

Comment:
Similar to the previous comment, the state updates for `byLengthPressed` and `alphabeticallyPressed` should not be inside the `sort` function. Move these updates outside to ensure they are handled correctly.